craft_store/store_client.py

```diff
@@ -68,9 +67,10 @@ def _wait_for_token(self, ctx, wait_token_url):
         )
 
 
-class StoreClient(HTTPClient):
+class StoreClient(BaseClient):
     """Encapsulates API calls for the Snap Store or Charmhub."""
 
+    @overrides
```

Comment:
We should use this more frequently in other projects for the extra verifications and to avoid docstring duplication.